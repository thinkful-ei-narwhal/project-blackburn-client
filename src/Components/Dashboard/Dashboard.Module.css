.dashboard-header {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-evenly;
    background-color: lightgray;
}

.links {
    text-decoration: none;
    font-size: 14px;
    color: #818181;
    padding: 50px;
}

.links:hover {
    text-decoration: none;
    font-size: 14px;
    color: #f1f1f1;
    cursor: pointer;
}

.links-selected {
    text-decoration: none;
    font-size: 14px;
    color: #f1f1f1;
    padding: 50px;
}

.dashboard-header-open {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: center;

    background-color: lightgray;

    margin-left: 25%;
}

.title {
    padding-left: 30px;
}

.sidenav {
    height: 100%;
    width: 0;
    position: fixed;
    z-index: 1;
    top: 0;
    left: 0;
    background-color: #111;
    overflow-x: hidden;
    transition: 0.5s;
    padding-top: 60px;
}

.sidenav-open {
    height: 100%;
    width: 25%;
    position: fixed;
    z-index: 1;
    top: 0;
    left: 0;
    background-color: #111;
    overflow-x: hidden;
    transition: 0.5s;

    display: flex;
    flex-direction: column;
    justify-content: space-evenly;
    align-items: center;
}

.sidenav-open a {
    text-decoration: none;
    font-size: 14px;
    color: #818181;
}

.navLinks a {   
    padding: 50px;
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    align-items: flex-start;
}

.x {
    color: red;
    font-size: 20px;
    cursor: pointer;
}


.content {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.content-open {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    margin-left: 30%;
}