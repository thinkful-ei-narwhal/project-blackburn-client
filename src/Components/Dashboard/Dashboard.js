import React from 'react';
import { Link } from 'react-router-dom';
import Leaderboard from '../Leaderboard/Leaderboard';
import Start from '../Start/Start';
import Analytics from '../Analytics/Analytics';
import Settings from '../Settings/Settings';
import './Dashboard.Module.css';
import BlackBurnContext from '../../Context/BlackburnContext';
import ScoreboardApiService from '../../Services/scoreboard-api-service';

export default class Dashboard extends React.Component {
  state = {
    menuOpen: true,
    showAnalytics: false,
    showLeaderboard: false,
    showSettings: false,
    showHome: true,
  };

  handleMenuButton = () => {
    if (!this.state.menuOpen) {
      this.setState({ menuOpen: true });
    } else {
      this.setState({ menuOpen: false });
    }
  };

  handleShowHome = () => {
    if (!this.state.showHome) {
      this.setState({
        showHome: true,
        showLeaderboard: false,
        showAnalytics: false,
        showSettings: false,
      });
    }
  };

  handleMenuButton = () => {
    if (!this.state.menuOpen) {
      this.setState({ menuOpen: true });
    } else {
      this.setState({ menuOpen: false });
    }
  };

  handleShowHome = () => {
    if (!this.state.showHome) {
      this.setState({
        showHome: true,
        showLeaderboard: false,
        showAnalytics: false,
        showSettings: false,
      });
    }
  };

  handleShowLeaderboard = () => {
    if (!this.state.showLeaderboard) {
      this.setState({
        showHome: false,
        showLeaderboard: true,
        showAnalytics: false,
        showSettings: false,
      });
    }
  };

  handleShowAnalytics = () => {
    if (!this.state.showAnalytics) {
      this.setState({
        showHome: false,
        showLeaderboard: false,
        showAnalytics: true,
        showSettings: false,
      });
    }
  };

  handleShowSettings = () => {
    if (!this.state.showSettings) {
      this.setState({
        showHome: false,
        showLeaderboard: false,
        showAnalytics: false,
        showSettings: true,
      });
    }
  };

  componentDidMount() {
    console.log(this.context);
    ScoreboardApiService.getAllScores('all').then((res) =>
      res.map((data) => {
        console.log(data);
        return this.setState({
          allScores: [
            ...this.state.allScores,
            {
              username: data.username,
              score: data.score,
              storyId: data.story_id,
            },
          ],
        });
      })
    );
    ScoreboardApiService.getMyScores(1, 'myscores').then((res) =>
      res.map((data) => {
        return this.setState({
          myScores: [
            ...this.state.myScores,
            { score: data.score, wpm: data.wpm, date: data.date_created },
          ],
        });
      })
    );
  }

  render() {
    return (
      <>
        <header
          className={
            this.state.menuOpen ? 'dashboard-header-open' : 'dashboard-header'
          }
        >
          {this.state.menuOpen ? (
            <div></div>
          ) : (
            <div onClick={() => this.handleMenuButton()}> &#9776; </div>
          )}
          <h1 className="title">Project Blackburn</h1>
          <Link to={'/'} className="links">
            {' '}
            logout{' '}
          </Link>
        </header>
        <div className={this.state.menuOpen ? 'sidenav-open' : 'sidenav'}>
          {this.state.menuOpen ? (
            <div className="x" onClick={() => this.handleMenuButton()}>
              {' '}
              &times;{' '}
            </div>
          ) : (
            <div></div>
          )}
          <nav className="navLinks">
            <div
              className={this.state.showHome ? 'links-selected' : 'links'}
              onClick={() => this.handleShowHome()}
            >
              {' '}
              Home{' '}
            </div>
            <div
              className={
                this.state.showLeaderboard ? 'links-selected' : 'links'
              }
              onClick={() => this.handleShowLeaderboard()}
            >
              {' '}
              Leaderboard{' '}
            </div>
            <div
              className={this.state.showAnalytics ? 'links-selected' : 'links'}
              onClick={() => this.handleShowAnalytics()}
            >
              {' '}
              Analytics{' '}
            </div>
            <div
              className={this.state.showSettings ? 'links-selected' : 'links'}
              onClick={() => this.handleShowSettings()}
            >
              {' '}
              Settings{' '}
            </div>
          </nav>
        </div>
        <div className={this.state.menuOpen ? 'content-open' : 'content'}>
          {this.state.showHome ? (
            <div>
              {' '}
              <Start />{' '}
            </div>
          ) : (
            <div></div>
          )}
          {this.state.showLeaderboard ? (
            <div>
              {' '}
              <Leaderboard allScores={this.state.allScores} />{' '}
            </div>
          ) : (
            <div></div>
          )}
          {this.state.showAnalytics ? (
            <div>
              {' '}
              <Analytics myScores={this.state.myScores} />{' '}
            </div>
          ) : (
            <div></div>
          )}
          {this.state.showSettings ? (
            <div>
              {' '}
              <Settings />{' '}
            </div>
          ) : (
            <div></div>
          )}
        </div>
      </>
    );
  }
}
