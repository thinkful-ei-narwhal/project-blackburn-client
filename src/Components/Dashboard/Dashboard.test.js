import React from 'react';
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Dashboard from './Dashboard';
import BlackBurnContext from '../../Context/BlackburnContext';

describe(`Adds dashboard to dom without crashing`, () => {
    it('renders without crashing', () => {
        const wrapper = shallow(
        <BlackBurnContext.Provider>
            <Dashboard />
        </BlackBurnContext.Provider>
    )
    .find('.dashboard-header-open')
    expect(toJson(wrapper)).toMatchSnapshot()
    })
})