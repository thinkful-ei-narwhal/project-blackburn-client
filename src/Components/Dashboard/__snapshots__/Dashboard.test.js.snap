// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adds dashboard to dom without crashing renders without crashing 1`] = `null`;
